modules/acquia_cms_common/src/EventSubscriber/KernelTerminate/AcquiaConnectorTelemetryOverride.php

```diff
@@ -25,7 +25,7 @@ class AcquiaConnectorTelemetryOverride extends AcquiaTelemetry {
   public function onTerminateResponse(KernelEvent $event) {
     // Check if telemetry opted,
     // then only trigger AcquiaTelemetry::onTerminateResponse($event).
-    if ($this->state->get('acquia_connector.telemetry.opted', TRUE)) {
+    if (PHP_SAPI !== 'cli' && \Drupal::service('state')->get('acquia_connector.telemetry.opted', TRUE)) {
```

Comment:
@rajeshreeputra Can we use DI here. 
In `AcquiaCmsCommonServiceOverrider` class, probably we can pass arguments something like:
```
->addArgument(new Reference('state'))
```
and use it here.